GPU/Vulkan/TextureCacheVulkan.cpp

```diff
@@ -652,12 +653,24 @@ void TextureCacheVulkan::ApplyTextureFramebuffer(TexCacheEntry *entry, VirtualFr
 		VkBuffer pushed;
 		uint32_t offset = push_->PushAligned(verts, sizeof(verts), 4, &pushed);
 
-		draw_->BindFramebufferAsTexture(framebuffer->fbo, 0, Draw::FB_COLOR_BIT, 0);
+		draw_->BindFramebufferAsTexture(framebuffer->fbo, 0, depth ? Draw::FB_DEPTH_BIT : Draw::FB_COLOR_BIT, 0);
 		VkImageView fbo = (VkImageView)draw_->GetNativeObject(Draw::NativeObject::BOUND_TEXTURE0_IMAGEVIEW);
 
 		VkDescriptorSet descSet = vulkan2D_->GetDescriptorSet(fbo, samplerNearest_, clutTexture->GetImageView(), samplerNearest_);
 		VulkanRenderManager *renderManager = (VulkanRenderManager *)draw_->GetNativeObject(Draw::NativeObject::RENDER_MANAGER);
 		renderManager->BindPipeline(depalShader->pipeline);
+
+		if (depth) {
+			DepthScaleFactors scaleFactors = GetDepthScaleFactors();
```

Comment:
With accurate depth off, we actually deform the stored depth (it incorrectly depends on the depth viewport parameters when drawing - much like the XY viewport problem), which might cause problems for depth texturing.  I think accurate depth is always on for Vulkan, though.

-[Unknown]

Reply:
Yeah we really need to get away from the stored depth being dependent on the viewport.. If we distort the viewport Z and the projection matrix together we should be able to get consistent results in the depth buffer while choosing how the clip happens, right?

Reply:
Well, accurate depth already does it "right".  It basically does as you describe, but also tries to buffer things to make depth clamp scenarios fail less often.  If we didn't distort depth, we'd have the same cases of depth going wrong that accurate depth has.

For example, that game where it uses a crazy depth scale (like a million), without incorrectly deforming the depth and without clamping, won't draw.  That's why it seems like accurate depth is "wrong" there, only because clamping isn't working and the deformed depth path is accidentally working.

-[Unknown]